cpp/dolfinx/fem/FiniteElement.h

```diff
@@ -81,10 +81,17 @@ class FiniteElement
   /// @return The string of the finite element family
   std::string family() const noexcept;
 
-  /// Evaluate all basis functions at given points in reference cell
-  /// reference_values[num_points][num_dofs][reference_value_size]
-  void evaluate_reference_basis(xt::xtensor<double, 3>& values,
-                                const xt::xtensor<double, 2>& X) const;
+  /// Evaluate all derivatives of the basis functions up to given order at given
+  /// points in reference cell
+  /// @param[in,out] values Four dimensional xtensor that will be filled with
+  /// the tabulated values. Should be of shape {num_derivatives, num_points,
+  /// num_dofs, reference_value_size}
+  /// @param[in] X Two dimensional xtensor of shape [num_points, geometric
+  /// dimension] containing the points at the reference element
+  /// @param[in] order The number of derivatives (up to and including this
+  /// order) to tabulate for.
+  void tabulate(xt::xtensor<double, 4>& values, const xt::xtensor<double, 2>& X,
```

Comment:
Should we also change the order of parameters in `CoordinateElement::tabulate` for consistency?